File: loja/pedidos/src/main/java/br/com/bota/queue/ProdutoQueue.java
package br.com.bota.queue;

import br.com.bota.dto.EstoqueDTO;
import io.micronaut.rabbitmq.annotation.Binding;
import io.micronaut.rabbitmq.annotation.RabbitClient;

//value na annotation rabbitClient, é o exchange.
@RabbitClient("exchange_baixa_estoque")
public interface ProdutoQueue {
    //route key
    @Binding("br.com.bota.produto")
    void send(EstoqueDTO estoqueDTO);
}

File: loja-kotlin/kpedidos/src/main/kotlin/br/com/bota/queue/ProdutoQueue.kt
package br.com.bota.queue

import br.com.bota.lojalib.dto.EstoqueDTO
import io.micronaut.rabbitmq.annotation.Binding
import io.micronaut.rabbitmq.annotation.RabbitClient

//value na annotation rabbitClient, é o exchange.
@RabbitClient("exchange_baixa_estoque")
interface ProdutoQueue {
    //route key
    @Binding("br.com.bota.produto")
    fun send(estoqueDTO: EstoqueDTO)
}